### PR TITLE
MHV-60728 - Add dd tags to secure-messaging web app

### PR DESCRIPTION
**Pull request by vsaleem**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [ ] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Implement missing and non-existing datadog tags within Secure Messaging app

## Related issue(s)

- [[MHV-60728]](https://jira.devops.va.gov/browse/MHV-60728) Implement Datadog tags

## Testing done

- [ ] No manual or automation testing required
- [ ]  Compared updates with `SM_Datadog.xlsx` spreedsheet
- [ ] 
## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/components/MessageThread/MessageThread.jsx [hunk 1]
```diff
@@ -41,6 +42,14 @@ const MessageThread = props => {
...
           data-testid="thread-expand-all"
           bordered
+          onClick={e => {
+            setExpanded(
+              e.target?.shadowRoot
+                ?.querySelector('.usa-accordion')
+                ?.querySelector('button').innerText,
+            );
+          }}
+          data-dd-action-name={`${expanded} Accordion Button`} //   `${e.
```
Comment (vsaleem) [inline src/applications/mhv-secure-messaging/components/MessageThread/MessageThread.jsx:52]: note to self, remove comment here

**Review by KyleCardwell — DISMISSED** [review]
Awesome!  Good work using the dataDogRum actions on all the close events 😄

**Review by KolbyVA — DISMISSED** [review]

File: src/applications/mhv-secure-messaging/components/MessageThread/MessageThreadItem.jsx [hunk 2]
```diff
@@ -103,6 +105,14 @@ const MessageThreadItem = props => {
...
           : `expand-message-button-${messageId}`
       }
+      onClick={e => {
+        setIsItemExpanded(e.target?.getAttribute('open'));
+      }}
+      data-dd-action-name={`${
+        isitemExpanded
+          ? 'Accordion Expanded Message'
+          : 'Accordion Collapsed Message'
+      }`}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/MessageThread/MessageThreadItem.jsx:115]: What do you think about calling `getAttribute` in `data-dd-action-name` directly to potentially avoid unnecessary rerenders?
```suggestion
      data-dd-action-name={`${
        e.target?.getAttribute('open')
          ? 'Accordion Expanded Message'
          : 'Accordion Collapsed Message'
      }`}

```

**Review by alex-morgun-1 — COMMENTED** [review]
there is one suggestion I made in `MessageThreadItem.jsx`. Other than that - all code looks good. I think this is a great addition to analytics. Thanks @vsaleem

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**